testpilot/users/views.py

```diff
@@ -38,5 +43,15 @@ def list(self, request):
         })
 
 
+@api_view(['POST'])
+@permission_classes([IsAuthenticated])
+def user_retire(request):
+    user = request.user
+    result = {"id": user.id, "username": user.username}
+    logout(request)
+    user.delete()
```

Comment:
Survey results will be in a different system altogether (SurveyMonkey, #568) and feedback is getting removed from the project (#567). Beyond that, [the list of what gets deleted is checked in this test](https://github.com/mozilla/testpilot/pull/614/files#diff-47c4478f8c1a25865ea8b4b6840d3163R459)